### ISSUE TITLE
play streams directly in CasparCG

### ISSUE THREAD

**Issue by vfmradio** [jaskie/PlayoutAutomation#63]
> You may also try with playing streams directly in CasparCG using i.e.
PLAY 1-0 STREAM "udp://127.0.0.1:56221" command.

Hello! Can anyone tell me why this is not working?

**Comment by jaskie** [comment]
TL;DR: PLAY 1-0 STREAM command does not work in our fork, because is not implemented.

It *may* work in [main trunk of CasparCG](https://github.com/CasparCG/server). Our fork is designed to be as stable as possible in MCR play-out, and typically work more than a year without restart. Main problem with playing anything other than local files is long time required to initialize source, related to different reasons, but almost always unacceptable in production. The only live producer that is sometimes good enough to use, it's Decklink input, it requires about 5 frames to synchronize. Well known workaround is to use separate CasaparCG channel for inputs, open source in advance, and play the second channel to main channel using channel producer (e.g. `PLAY 1-2 route://2`). In this way you can use NDI producer (`PLAY 2-0 NDI COMPUTER_NAME\SOURCE_NAME`) from our fork.
Second problem is stream synchronization - it's absent in 2.06, in our fork there is only initial synchronization, and probably still not available in main trunk. The synchronization for incoming UDP stream is crucial, as in most DVB streams audio is ahead of video at least 0.5 second. Without complete sync this kind of stream producer seems to be useless.

**Comment by jaskie** [comment]
Just wrote [FAQ](https://github.com/jaskie/PlayoutAutomation/wiki/Faq) on it.
